File: src/bnl-dark-api/bnl-dark-api/Models/Therapy.cs
namespace bnl_dark_api.Models;

public class Therapy : ITherapy, IId
{
    public int Id { get; set; }
    public int Amount { get; set; }
    public int TimesTillRecovered { get; set; }
    public IMedicine Medicine { get; set; }
}

public interface ITherapy
{
    public int Id { get; set; }
    public int Amount { get; set; }
    public int TimesTillRecovered { get; set; }
    public IMedicine Medicine { get; set; }
}

File: src/bnl-dark-web/src/app/patient-management/models/therapy.ts
import { IMedicine } from '../../master-data-management/models/medicine';
import { IStay } from './stay';

export interface ITherapy {
  id: number;
  timesDone: number;
  totalTimesToBeHealed: number;
  medicineAmountPerIteration: number;
  medicine: IMedicine;
  medicineId: number;
  stay: IStay;
  stayId: number;
}
